pandas/core/internals/construction.py

```diff
@@ -202,15 +206,21 @@ def mgr_to_mgr(mgr, typ: str):
         if isinstance(mgr, BlockManager):
             new_mgr = mgr
         else:
-            new_mgr = arrays_to_mgr(
-                mgr.arrays, mgr.axes[0], mgr.axes[1], mgr.axes[0], typ="block"
-            )
+            if mgr.ndim == 2:
```

Comment:
can you make elif/else

Reply:
I personally wouldn't find that an improvement in readability. Now it's very clearly separated into two cases (manager already has the same type -> use as is, or else the manager needs to be converted)

Reply:
using nested elif/if is an anti-pattern

Reply:
It's only an anti-pattern if it's deeply nested and hampering readability. In this case, it makes the code easier to understand IMO.